### PR TITLE
Dependencies not shown when a task doesn't have a decision task

### PR DESCRIPTION
**Pull request by helfi92**
Fixes https://bugzilla.mozilla.org/show_bug.cgi?id=1595418 and https://bugzilla.mozilla.org/show_bug.cgi?id=1595506.

### CODE REVIEW

File: services/web-server/src/loaders/tasks.js [hunk 1]
```diff
@@ -3,16 +3,15 @@ const sift = require('../utils/sift');
...
-      taskIds.map(async taskId => {
-        const task = await queue.task(taskId);
+    Promise.all(taskIds.map(async (taskId) => {
+      const [error, task] = await tryCatch(queue.task(taskId));
 
-        return new Task(taskId, null, task);
-      }),
-    ),
+      return error || new Task(taskId, null, task);
+    })),
```
Comment (helfi92) [inline services/web-server/src/loaders/tasks.js]: We would need to do a similar logic to all loaders that use `Promise.all`. I could take care of that later, in a separate PR.

File: services/web-server/src/loaders/tasks.js [hunk 2]
```diff
@@ -3,16 +3,15 @@ const sift = require('../utils/sift');
...
-    Promise.all(
-      taskIds.map(async taskId => {
-        const task = await queue.task(taskId);
+    Promise.all(taskIds.map(async (taskId) => {
+      const [error, task] = await tryCatch(queue.task(taskId));
 
-        return new Task(taskId, null, task);
-      }),
-    ),
+      return error || new Task(taskId, null, task);
```
Comment (djmitche) [inline services/web-server/src/loaders/tasks.js]: Will returning an Error object do the right thing, or would this need to be thrown?

Comment (helfi92) [reply services/web-server/src/loaders/tasks.js]: > Will returning an Error object do the right thing

Good question. The dataloader package seems to check for the instance of the object so we are good.

https://github.com/graphql/dataloader/blob/efc2b1a376863daeee88ff8e3bc901bb54fc5356/src/index.js#L293

**Comment by helfi92** [comment]
> I don't really understand how this works. I think I don't understand when there would be multiple taskIds

https://www.apollographql.com/docs/graphql-tools/connectors/#dataloader-and-caching has a good explanation.

File: services/web-server/src/loaders/tasks.js (hunk 2 above)
Comment (helfi92) [reply services/web-server/src/loaders/tasks.js]: Good question. The dataloader package seems to check for the instance of the object so we are good.

https://github.com/graphql/dataloader/blob/efc2b1a376863daeee88ff8e3bc901bb54fc5356/src/index.js#L293

**Pull request merged**